reference/shaders-msl/vert/basic.capture.vert

```diff
@@ -0,0 +1,35 @@
+#include <metal_stdlib>
+#include <simd/simd.h>
+
+using namespace metal;
+
+struct UBO
+{
+    float4x4 uMVP;
+};
+
+struct spvAux
+{
+    uint vertexCount;
+    uint swizzleConst[1];
+};
+
+struct main0_out
+{
+    float3 vNormal [[user(locn0)]];
+    float4 gl_Position [[position]];
+};
+
+struct main0_in
+{
+    float4 aVertex [[attribute(0)]];
+    float3 aNormal [[attribute(1)]];
+};
+
+vertex void main0(main0_in in [[stage_in]], constant UBO& _16 [[buffer(0)]], constant spvAux& spvAuxBuffer [[buffer(30)]], uint gl_VertexIndex [[vertex_id]], uint gl_BaseVertex [[base_vertex]], uint gl_InstanceIndex [[instance_id]], uint gl_BaseInstance [[base_instance]], device main0_out* spvOut [[buffer(29)]])
+{
+    device main0_out& out = spvOut[(gl_InstanceIndex - gl_BaseInstance) * spvAuxBuffer.vertexCount + gl_VertexIndex - gl_BaseVertex];
```

Comment:
Have you tested this with nested functions? If we need to pass down global variables to leaf function, they'll now need to have device storage space and not thread.

Reply:
This actually occurred to me, which is why I added code to `get_argument_address_space()` to handle this case. I'll add a test to make sure this is actually working.

Reply:
Right, I thought I saw that, just wanted to make sure.